cpp/arcticdb/stream/stream_reader.hpp

```diff
@@ -116,29 +116,32 @@ class StreamReader {
     }
 
     auto generate_rows() {
-        return folly::gen::from(key_gen_())
-            | generate_segments_from_keys(*store_, IDX_PREFETCH_WINDOW, opts_)
-            | generate_keys_from_segments(*store_, entity::KeyType::TABLE_DATA, entity::KeyType::TABLE_INDEX)
-            | generate_segments_from_keys(*store_, DATA_PREFETCH_WINDOW, opts_)
-            | generate_rows_from_data_segments();
+        auto keys = key_gen_();
```

Comment:
It looks like the previous version was lazy evaluated. Can you verify that this new version is also lazy?